WonSeungHyun/app.js

```diff
@@ -37,3 +35,118 @@ const start = async () => {
 };
 
 start();
+
+app.post("/users", async (req, res, next) => {
+  const { name, email, profile_image } = req.body;
+
+  await appDataSource.query(
+    `
+    INSERT INTO users(
+    name,
+    email,
+    profile_image
+    
+  )VALUES (?, ?, ?)`,
+    [name, email, profile_image]
+  );
+  res.status(201).json({ message: "successfully created" });
+});
+
+app.get("/search", async (req, res) => {
+  await appDataSource.query(
+    `SELECT
+     u.id as userId,
+     u.profile_image as userProfileImage,
+     p.id as PostingId,
+     
+     p.content as PostingContent
+     FROM posts p, users u
+     WHERE p.user_id = u.id`,
+    (err, rows) => {
+      res.status(200).json(rows);
+    }
+  );
+});
```

Comment:
RESTful api를 구현하는데 있어 몇가지 주의할 점이 있습니다.

엔드포인트에는 동사형이 아닌 명사형으로 작성해주어야 합니다.

이미 HTTP 메서드인 GET은 어떤 리소스를 조회하겠다는 의미가 있으므로, 엔드포인트를 다음과 같이 수정해주세요.

추가로, FROM 절에서 여러 테이블을 한번에 불러오는 방식이 아닌 JOIN 결합을 통해 특정 정보를 가져올 수 있도록 수정해주세요!

```suggestion
app.get("/posts", async (req, res) => {
  await appDataSource.query(
    `SELECT
       u.id as userId,
       u.profile_image as userProfileImage,
       p.id as postingId,
       p.title as postingTitle,
       p.content as postingContent
     FROM posts p
     INNER JOIN users u ON u.id = p.user_id
     WHERE p.user_id = u.id`,
    (err, rows) => {
      res.status(200).json(rows);
    }
  );
});
```